scanspec/core.py

```diff
@@ -176,23 +176,31 @@ def if_instance_do(x, cls: Type, func: Callable):
         return NotImplemented
 
 
+def last_first_gap(upper: AxesPoints, lower: AxesPoints) -> bool:
```

Comment:
Not sure about this function name for readability purposes. How about `is_gap_between`?

Reply:
Maybe `is_gap_between(dim1, dim2)`